Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponentSlot.cpp

```diff
@@ -311,75 +291,106 @@ namespace AZ
             action->setEnabled(HasSourceData());
 
-            action = menu.addAction("Edit Material Instance...", [this]() { OpenMaterialInspector(); });
-            action->setEnabled(GetActiveAssetId().IsValid());
+            action = menu.addAction("Edit Material Instance...", [this, entityIdsToEdit]() { OpenMaterialInspector(entityIdsToEdit); });
+            action->setEnabled(GetActiveAssetId().IsValid() && hasMatchingMaterialTypes);
 
-            action = menu.addAction("Edit Material Instance UV Map...", [this]() { OpenUvNameMapInspector(); });
-            action->setEnabled(GetActiveAssetId().IsValid());
+            action = menu.addAction("Edit Material Instance UV Map...", [this, entityIdsToEdit]() { OpenUvNameMapInspector(entityIdsToEdit); });
+            action->setEnabled(GetActiveAssetId().IsValid() && hasMatchingMaterialTypes);
 
             menu.addSeparator();
 
-            MaterialPropertyOverrideMap propertyOverrides;
-            MaterialComponentRequestBus::EventResult(
-                propertyOverrides, m_entityId, &MaterialComponentRequestBus::Events::GetPropertyOverrides, m_id);
-            RPI::MaterialModelUvOverrideMap matModUvOverrides;
-            MaterialComponentRequestBus::EventResult(
-                matModUvOverrides, m_entityId, &MaterialComponentRequestBus::Events::GetModelUvOverrides, m_id);
+            action = menu.addAction("Clear Material Instance Overrides", [this, entityIdsToEdit]() {
+                for (const AZ::EntityId& entityId : entityIdsToEdit)
+                {
+                    MaterialComponentRequestBus::Event(
+                        entityId, &MaterialComponentRequestBus::Events::SetPropertyOverrides, m_id, MaterialPropertyOverrideMap());
+                    MaterialComponentRequestBus::Event(
+                        entityId, &MaterialComponentRequestBus::Events::SetModelUvOverrides, m_id,
+                        AZ::RPI::MaterialModelUvOverrideMap());
+                }
+                OnDataChanged(entityIdsToEdit, false);
+            });
 
-            action = menu.addAction("Clear Material Instance Overrides", [this]() { ClearOverrides(); });
-            action->setEnabled(!propertyOverrides.empty() || !matModUvOverrides.empty());
+            action = menu.addAction("Clear Material And Properties", [this, entityIdsToEdit]() {
+                m_materialAsset = {};
+                for (const AZ::EntityId& entityId : entityIdsToEdit)
+                {
+                    MaterialComponentRequestBus::Event(
+                        entityId, &MaterialComponentRequestBus::Events::SetPropertyOverrides, m_id, MaterialPropertyOverrideMap());
+                    MaterialComponentRequestBus::Event(
+                        entityId, &MaterialComponentRequestBus::Events::SetModelUvOverrides, m_id,
+                        AZ::RPI::MaterialModelUvOverrideMap());
+                }
+                OnDataChanged(entityIdsToEdit, true);
+            });
```

Comment:
Do these need ScopedUndoBatch?

Reply:
All of that is handled inside of OnDataChanged